src/partition/ReceivedPartition.cpp

```diff
@@ -470,143 +488,312 @@ void ReceivedPartition::createOwnerInformation()
   PRECICE_TRACE();
   Event e("partition.createOwnerInformation." + _mesh->getName(), precice::syncMode);
 
-  if (utils::MasterSlave::isSlave()) {
+  if (m2n().usesTwoLevelInitialization()) {
+    /*
+    1- receive local bb map from master
+    2- filter bb map to keep the connected ranks
+    3- own the vertices that only fits into the ranks bb
+    4- send number of owned vertices and shared vertices list to neighbors
+    5- for the remaining vertices: check if we have less vertexes -> own it!
+    */
+
+    // #1: receive local bb map from master
+    // Define and initialize localBBMap to save local bbs
+    prepareBoundingBox();
+   
+    mesh::Mesh::BoundingBoxMap localBBMap;
+    for (int rank = 0; rank < utils::MasterSlave::getSize(); rank++) {
+      localBBMap.emplace(rank, _bb);
+    }
+
+    // Define a bb map to save the local connected ranks and respective boundingboxes 
+    mesh::Mesh::BoundingBoxMap localConnectedBBMap;
+
+    // store global IDs and list of possible shared vertices
+
+    // Global IDs to be saved in: 
+    std::vector<int> sharedVerticesList;
+    // Local IDs to be saved in:
+    std::vector<int> sharedVerticesIndex;
+    
+    // store possible shared vertices in a map to communicate with neighbors, map: rank -> vertex_global_id
+    std::map<int, std::vector<double>> sharedVerticesSendMap;
+
+    // receive list of possible shared vertices from neighboring ranks
+    std::map<int, std::vector<double>> sharedVerticesReceiveMap;
+
+    if (utils::MasterSlave::isMaster()) {
+
+      // master receives local bb from each slave ranke
+      for (int rankSlave = 1; rankSlave < utils::MasterSlave::getSize(); rankSlave++) {
+        com::CommunicateBoundingBox(utils::MasterSlave::_communication).receiveBoundingBox(localBBMap.at(rankSlave), rankSlave);
+      }
+
+      // master broadcast localBBMap to all slaves 
+      for (int rankSlave = 1; rankSlave < utils::MasterSlave::getSize(); rankSlave++) {
+        com::CommunicateBoundingBox(utils::MasterSlave::_communication).sendBoundingBoxMap(localBBMap, rankSlave);
+      }
+    }
+    else if (utils::MasterSlave::isSlave()) {
+      // slaves send local bb to master 
+      com::CommunicateBoundingBox(utils::MasterSlave::_communication).sendBoundingBox(_bb, 0);
+      // slaves receive localBBMap from master
+      com::CommunicateBoundingBox(utils::MasterSlave::_communication).receiveBoundingBoxMap(localBBMap, 0);
+    }    
+
+    // #2: filter bb map to keep the connected ranks
+    // remove the current bb from the map
```

Comment:
```suggestion
    // remove the own bb from the map
```
please add an explanation why we do this